ssetests/tests_basic_parsing.go

```diff
@@ -1,19 +1,67 @@
 package ssetests
 
 import (
+	"crypto/rand"
+	"encoding/base64"
 	"fmt"
 	"time"
 
 	"github.com/launchdarkly/sse-contract-tests/framework/ldtest"
+
+	"github.com/stretchr/testify/assert"
 )
 
+func generateRandomString(length int) string {
+	b := make([]byte, length)
+	_, err := rand.Read(b)
+	if err != nil {
+		panic(err)
+	}
+	return base64.StdEncoding.EncodeToString(b)
+}
+
 func DoBasicParsingTests(t *ldtest.T) {
 	t.Run("one-line message in one chunk", func(t *ldtest.T) {
 		_, stream, client := NewStreamAndSSEClient(t)
 		stream.Send("data: Hello\n\n")
 		client.RequireSpecificEvents(t, EventMessage{Data: "Hello"})
 	})
 
+	t.Run("two messages spanning 3 chunks with shared chunk", func(t *ldtest.T) {
+		// This test primarily tests situations where the implementation over allocates buffers to decrease the total
+		// number of buffers. This test helps ensure that the used size in the buffer is properly tracked.
+		_, stream, client := NewStreamAndSSEClient(t)
+		stream.Send("data: test")
+		stream.Send("test\n\ndata:")
+		stream.Send("test" + "\n\n")
+		client.RequireSpecificEvents(t, EventMessage{Data: "testtest"})
+		client.RequireSpecificEvents(t, EventMessage{Data: "test"})
+	})
+
+	t.Run("large message in one chunk", func(t *ldtest.T) {
+		_, stream, client := NewStreamAndSSEClient(t)
+		randomData := generateRandomString(5 * 1024 * 1024)
+		stream.Send("data: " + randomData + "\n\n")
+		actual := client.RequireEvent(t)
+		// Does not use RequireSpecificEvents, because then it would print megabytes of text.
+		if actual.Data != (randomData) {
```

Comment:
One thing you could do is print out the seed used to generate the data, and the amount, and then you could reproduce it exactly when debugging.